src/vanilla/utils/proxyMap.ts

```diff
@@ -1,111 +1,152 @@
-import { proxy } from '../../vanilla.ts'
+import { getVersion, proxy } from '../../vanilla.ts'
 
-type KeyValRecord<K, V> = [key: K, value: V]
+const maybeProxify = (x: any) => proxy({ x }).x
 
-type InternalProxyMap<K, V> = Map<K, V> & {
-  data: KeyValRecord<K, V>[]
-  toJSON: object
+const isSnapshot = (context: any) => getVersion(context) === undefined
```

Comment:
Nit: It's fine in this context, but to be precise we should reverse it to avoid misleading.
```js
const isProxy = (x: any) => getVersion(x) !== undefined
```

Reply:
Aren't snapshots technically proxies as well? How about `isImmutable`?

Reply:
No. In the context of `valtio/vanilla`, snapshot(...) returns a plain object.

I think you are talking about `useSnapshot()` returning a proxy object for tracking. We don't usually call it a "valtio" proxy. Yeah, it's confusing anyways.

I'm fine with `isSnapshot`. Maybe just leave a comment saying, this is an internal use and not general purpose. My concern was people wrongly assumes `isSnapshot` is a valid technique to know if "any" object is a snapshot.